src/App.tsx

```diff
@@ -16,7 +16,7 @@ function App() {
   const dispatch = useDispatch();
   useEffect(() => {
     dispatch(getPosts());
-  }, []);
+  });
```

Comment:
we actually want the empty array of dependencies, otherwise `getPosts()` will be dispatched whenever the component updates, i.e. when theme changes
```suggestion
  }, []);
```